src/BalanceLaws/interface.jl

```diff
@@ -1,22 +1,62 @@
 #### Balance Law Interface
-
 """
-    BalanceLaw
+    abstract type BalanceLaw end
 
 An abstract type representing a PDE balance law of the form:
 
 ```math
-q_{,t} + Σ_{i=1,...d} F_{i,i} = s
+\\frac{dq}{dt} = \\nabla \\cdot F_1(q, a, t) + \\nabla \\cdot F_2(q, \\nabla g, h, a, t) + S(q, \\nabla g, a, t)
 ```
+where:
+- ``q`` is the prognostic state,
+- ``a`` is the auxiliary state,
+- ``g = G(q, a, t)`` is the gradient state (variables of which we compute the
+  gradient),
+- ``h`` is the hyperdiffusive state.
+
+Subtypes of `BalanceLaw` should define the following interfaces:
+- [`vars_state`](@ref) to define the prognostic, auxiliary and intermediate
+  variables.
+- [`flux_first_order!`](@ref) to compute ``F_1``
+- [`flux_second_order!`](@ref) to compute ``F_2``
+- [`source!`](@ref) to compute ``S``
+
+If `vars(bl, ::GradientFlux, FT)` is non-empty, then the following should be
+defined:
+- [`compute_gradient_argument!`](@ref) to compute ``G``
+- [`compute_gradient_flux!`](@ref) is a linear transformation of ``\nabla g``
```

Comment:
```suggestion
- [`compute_gradient_flux!`](@ref) is a linear transformation of ``∇g``
```
At the moment, `\n` is parsed as a newline.

Reply:
Good catch. Thanks!